File: packages/backend/schemas/catalog.py
from pydantic import BaseModel


class CatalogBestProvider(BaseModel):
    name: str
    display_name: str
    avg_tps: float


class CatalogItem(BaseModel):
    model: str
    provider_count: int
    avg_tps: float
    avg_ttft_ms: int
    total_samples: int
    best_provider: CatalogBestProvider | None = None


class ModelsCatalogResponse(BaseModel):
    items: list[CatalogItem]
    total: int


class ModelProviderItem(BaseModel):
    provider: str
    provider_name: str
    avg_tps: float
    avg_ttft_ms: int
    sample_count: int
    data_quality: str


class ModelProvidersCatalogResponse(BaseModel):
    model: str
    providers: list[ModelProviderItem]


class TopModelItem(BaseModel):
    name: str
    avg_tps: float
    avg_ttft_ms: int
    sample_count: int


class ProviderCatalogItem(BaseModel):
    provider: str
    name: str
    logo_url: str
    model_count: int
    avg_tps: float
    avg_ttft_ms: int
    total_samples: int
    top_models: list[TopModelItem]


class ProvidersCatalogResponse(BaseModel):
    items: list[ProviderCatalogItem]
    total: int

File: packages/frontend/lib/types/catalog.ts
// 目录/列表相关类型

export type ModelCatalogItem = {
  model: string;
  provider_count: number;
  avg_tps: number;
  avg_ttft_ms: number;
  total_samples: number;
  best_provider: {
    name: string;
    display_name: string;
    avg_tps: number;
  };
};

export type ModelsCatalogResponse = {
  items: ModelCatalogItem[];
  total: number;
};

export type ModelProviderDetail = {
  provider: string;
  provider_name: string;
  model: string;
  avg_tps: number;
  avg_ttft_ms: number;
  sample_count: number;
  data_quality: "sufficient" | "limited";
};

export type ProviderTopModel = {
  name: string;
  avg_tps: number;
  avg_ttft_ms: number;
  sample_count: number;
};

export type ProviderCatalogItem = {
  provider: string;
  name: string;
  logo_url?: string;
  model_count: number;
  avg_tps: number;
  avg_ttft_ms: number;
  total_samples: number;
  top_models: ProviderTopModel[];
};

export type ProvidersCatalogResponse = {
  items: ProviderCatalogItem[];
  total: number;
};
